src/demo_main.cpp

```diff
@@ -10,26 +11,50 @@
 #include "surface_perception/surface_objects.h"
 #include "surface_perception/typedefs.h"
 #include "surface_perception/visualization.h"
+#include "tf/transform_listener.h"
+#include "tf_conversions/tf_eigen.h"
 
 using surface_perception::SurfaceObjects;
 using surface_perception::SurfaceViz;
 
 class Demo {
  public:
-  Demo(const SurfaceViz& viz, const ros::Publisher input_pub);
+  Demo(const SurfaceViz& viz, const std::string& target_frame,
+       const ros::Publisher& input_pub);
   void Callback(const sensor_msgs::PointCloud2ConstPtr& cloud);
 
  private:
   SurfaceViz viz_;
   ros::Publisher input_pub_;
+  std::string target_frame_;
+  tf::TransformListener tf_listener_;
 };
 
-Demo::Demo(const SurfaceViz& viz, const ros::Publisher input_pub)
-    : viz_(viz), input_pub_(input_pub) {}
+Demo::Demo(const SurfaceViz& viz, const std::string& target_frame,
+           const ros::Publisher& input_pub)
+    : viz_(viz),
+      target_frame_(target_frame),
```

Comment:
Can you move this below input_pub_? Generally, you should initialize members in the same order they were declared in the class.

Reply:
Completed

Reply:
In addition, README was updated to include demo instruction.